File: csharp/FindNestedControl.cs
//https://gist.github.com/727225
public static T FindNestedControl<T>(Control controlToSearch, string id) where T: Control
{
	T foundControl = null;
	bool exceptionThrown = false;

	// look directly in the control first
	try
	{
		foundControl = (T)controlToSearch.FindControl(id);
	}
	catch (InvalidCastException)
	{
		exceptionThrown = true;
	}

	// if the control wasn't found in the root control search all sub-controls for it
	if ((foundControl == null) || exceptionThrown)
	{
		foreach (Control subControl in controlToSearch.Controls)
		{
			foundControl = FindNestedControl<T>(subControl, id);
			if (foundControl != null)
			{
				break;
			}
		}
	}

	return foundControl;
}

// FindNestedControl(ctlEmployeeInformationForm, "FirstName").Text

File: visualbasic/FindNestedControl.vb
'https://gist.github.com/727225
Public Shared Function FindNestedControl(Of T As System.Web.UI.Control)(ByVal controlToSearch As System.Web.UI.Control, ByVal id As String) As T
	Dim foundControl As T = Nothing
	Dim exceptionThrown As Boolean = False

	' look directly in the control first
	Try
		foundControl = DirectCast(controlToSearch.FindControl(id), T)
	Catch generatedExceptionName As InvalidCastException
		exceptionThrown = True
	End Try

	' if the control wasn't found in the root control search all sub-controls for it
	If (foundControl Is Nothing) OrElse exceptionThrown Then
		For Each subControl As System.Web.UI.Control In controlToSearch.Controls
			foundControl = FindNestedControl(Of T)(subControl, id)
			If foundControl IsNot Nothing Then
				Exit For
			End If
		Next
	End If

	Return foundControl
End Function

' FindNestedControl(ctlEmployeeInformationForm, "FirstName").Text